textproc/lilypond/Portfile

```diff
@@ -17,47 +17,84 @@ long_description    Lilypond is a unix-based automated engraving system that \
-                    port:pkgconfig
-depends_lib         port:fontconfig \
-                    port:fontforge \
-                    port:freetype \
-                    port:gettext \
-                    port:gmp \
-                    path:lib/pkgconfig/glib-2.0.pc:glib2 \
-                    port:ghostscript \
-                    port:mftrace \
-                    port:guile18 \
                     port:texinfo \
-                    path:lib/pkgconfig/pango.pc:pango \
-                    port:flex \
-                    port:t1utils \
+                    port:texlive-fonts-recommended \
                     port:texlive-lang-cyrillic \
                     port:texlive-metapost \
-                    port:dblatex \
-                    port:urw-fonts \
-                    port:libtool \
-                    path:bin/Xquartz:xorg-server \
-                    port:python27
+                    port:texlive-xetex \
+                    port:urw-core35-fonts
 
-configure.python    ${frameworks_dir}/Python.framework/Versions/2.7/bin/python2.7
+depends_lib         port:extractpdfmark \
+                    port:ghostscript \
+                    port:guile18 \
+                    port:python38
+
+build.type          gnu
+configure.cmd       autoconf -f && ./configure
+configure.python    ${prefix}/bin/python3.8
+
+configure.args-append   --disable-documentation
+configure.args-append   --with-texgyre-dir=${prefix}/share/texmf-texlive/fonts/opentype/public/tex-gyre
+configure.args-append   --with-urwotf-dir=${prefix}/share/fonts/urw-core35-fonts
+
+configure.env-append    GUILE=${prefix}/bin/guile18
+configure.env-append    GUILE_CONFIG=${prefix}/bin/guile18-config
+configure.env-append    GUILE_TOOLS=${prefix}/bin/guile18-tools
+
+configure.env-append    PYTHON=${configure.python}
+build.env-append        PYTHON=${configure.python}
+destroot.env-append     PYTHON=${configure.python}
+
+configure.env-append    PYTHON_CONFIG=${configure.python}-config
+build.env-append        PYTHON_CONFIG=${configure.python}-config
+destroot.env-append     PYTHON_CONFIG=${configure.python}-config
+
+configure.env-append    LTDL_LIBRARY_PATH=${prefix}/lib
+build.env-append        LTDL_LIBRARY_PATH=${prefix}/lib
+destroot.env-append     LTDL_LIBRARY_PATH=${prefix}/lib
 
-patchfiles          patch-scripts-build-mf2pt1.pl.diff
 post-patch {
-    reinplace s|__vector|lily_vector|g ${worksrcpath}/flower/include/std-vector.hh
-    set files {Documentation/contributor/build-notes.itexi \
-        Documentation/cs/usage/setup.itely \
-        Documentation/de/usage/setup.itely \
+    # Use guile18 header files.
+    reinplace -W ${worksrcpath} \
+        s|libguile\.h|libguile18.h|g \
+        aclocal.m4 \
+        configure.ac \
+        lily/include/lily-guile.hh
+
+    # Correct mf2pt1 binary location.
+    reinplace -W ${worksrcpath} \
+        s|"-mem=mf2pt1"|"-mem=../../mf2pt1"|g \
+        scripts/build/mf2pt1.pl
+
+    # Correct python shebang.
```

Comment:
Do we need this?  It's not part of the `lilypond-devel` portfile.  If it is necessary I will update the development portfile accordingly.

Reply:
Are you asking about the shebang fixing? If so, AFAIK it is not (yet) needed in `lilypond-devel` because we are still using python2 there.

Reply:
Nope, `lilypond-devel` already uses python37.

Reply:
Then I believe it may be necessary to do something similar to get the python selection to reliably grab the correct version.

Reply:
Do we think it would be worth it to create python version variants?

Reply:
I'd stick to just one python.

Reply:
I would like to have the Portfiles for `lilypond` and `lilypond-devel` as similar as possible – now is the right time to do that IMHO.

As a consequence, I will eventually submit a push request to add the `+mactex` variant to `lilypond`; it's then trivial to copy the new Portfile to `lilypond-devel` with only very minor changes, and it will also contain the python selection stuff.